bench-tps/src/bench.rs

```diff
@@ -315,15 +308,10 @@ fn do_tx_transfers<T: Client>(
                 duration_as_ms(&transfer_start.elapsed()),
                 tx_len as f32 / duration_as_s(&transfer_start.elapsed()),
             );
-            solana_metrics::submit(
-                influxdb::Point::new("bench-tps")
-                    .add_tag("op", influxdb::Value::String("do_tx_transfers".to_string()))
-                    .add_field(
-                        "duration",
-                        influxdb::Value::Integer(duration_as_ms(&transfer_start.elapsed()) as i64),
-                    )
-                    .add_field("count", influxdb::Value::Integer(tx_len as i64))
-                    .to_owned(),
+            datapoint!(
```

Comment:
I was hoping we wouldn't have to export any macro other than `datapoint!`.   I'm pretty sure the rust macro support is expressive enough to define a syntax perhaps like this, so that the `field!` macro is unnecessary:

```rust
datapoint!(
                "bench-tps-do_tx_transfers",
                ("duration", duration_as_ms(&transfer_start.elapsed()), i64),
                ("count", tx_len, i64)
            );
```

Reply:
I'll take a look

Reply:
Have any existing examples in mind?

Reply:
Is Rust's `match` of types or tokens compile time?  If so then I I think we could do it.

Reply:
compile time.  The serde_json crate has a fun `json!()` macro that demos a custom language: https://docs.serde.rs/serde_json/macro.json.html

Reply:
heh, that's some serious fu, need lunch first :-)